Update documentation build docs

Update https://ml.dask.org/contributing.html#documentation to be more in line with https://docs.dask.org/en/latest/develop.html#contributing-to-documentation

In addition I just tried to run
```
conda env create -f ci/environment-docs.yaml --name=dask-ml-dev-docs
cd docs
make html
```
and got
```
(dask-ml-dev-docs) Ray@UM-404XFVH4 docs % make html
Running Sphinx v1.8.5

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/opt/anaconda3/envs/dask-ml-dev-docs/lib/python3.6/site-packages/sphinx/config.py", line 368, in eval_config_file
    execfile_(filename, namespace)
  File "/opt/anaconda3/envs/dask-ml-dev-docs/lib/python3.6/site-packages/sphinx/util/pycompat.py", line 150, in execfile_
    exec_(code, _globals)
  File "/Users/Ray/Documents/PYTHON_dev/dask-ml/docs/source/conf.py", line 23, in <module>
    from dask_ml import __version__ as version
ModuleNotFoundError: No module named 'dask_ml'

make: *** [html] Error 2
```

--- a/docs/source/contributing.rst
+++ b/docs/source/contributing.rst
@@ -143,6 +143,10 @@ Building the docs is possible with
 .. code-block:: none
 
    $ conda env create -f ci/environment-docs.yaml --name=dask-ml-dev-docs
+   $ conda activate dask-ml-dev-docs
+   $ python -m pip install -e .
+   $ cd docs
+   $ make html
 
 Examples are written as Jupyter notebooks with their output stripped, either
 manually or using `nbstripout <https://github.com/kynan/nbstripout>`_. We want

